src/Symfony/Component/DependencyInjection/Compiler/AbstractableArgumentPresenceExceptionPass.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Argument\AbstractableArgument;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+/**
+ * Throws an exception for any service that have an abstract argument.
+ */
+class AbstractableArgumentPresenceExceptionPass extends AbstractRecursivePass
```

Comment:
I think the pass is not needed because consumers of containers must deal with `AbstractableArgument` anyway.
ContainerBuilder is missing the same check as PhpDumper btw

Reply:
> I think the pass is not needed because consumers of containers must deal with `AbstractableArgument` anyway.

@nicolas-grekas, pass just need to be removed and the only checkpoints with exceptions are `ContainerBuilder` and `PhpDumper`?
it seemed to me that the main idea of abstract argument is to check at container compilation time and compiled container must not contain an abstract argument. or we can rely on exception in `PhpDumper` during compilation?

> ContainerBuilder is missing the same check as PhpDumper btw

`ContainerBuilder::doResolveServices()` looks like the only suitable for an exception, but it will not thrown at compilation time

Reply:
>  pass just need to be removed and the only checkpoints with exceptions are ContainerBuilder and PhpDumper?

correct

>  the main idea of abstract argument is to check at container compilation time

where the check is done is an implementation detail - actually eg XmlDumper doesn't want throwing to happen.
The point of abstract arguments is to allow defining arguments that must be defined in a DI extension or in a pass. The rest is implementation details.

Reply:
> ContainerBuilder::doResolveServices() looks like the only suitable for an exception, but it will not thrown at compilation time

correct - and we don't care about throwing at compile time :)

Reply:
fixed